src/plugins/controls/public/control_group/embeddable/control_group_container.tsx

```diff
@@ -197,6 +197,32 @@ export class ControlGroupContainer extends Container<
       })
     );
 
+    const arrayOfBehaviorSubjects: Array<BehaviorSubject<string | undefined>> =
+      this.getChildIds().map((childId) => {
+        const child = this.getChild(childId);
+        if (isClearableControl(child)) {
+          return child.invalidSelections$;
+        }
+        return new BehaviorSubject<string | undefined>(undefined);
+      });
+    this.subscriptions.add(
+      combineLatest(arrayOfBehaviorSubjects)
+        .pipe(
+          debounceTime(100),
+          map((stuff) => {
+            return stuff.filter((a) => a !== undefined);
+          }),
+          distinctUntilChanged((a, b) => isEqual(a, b))
+        )
+        .subscribe((childHasInvalidSelection) => {
+          if (childHasInvalidSelection.length > 0) {
+            this.dispatch.setInvalidSelectionsControlId(childHasInvalidSelection[0]);
+          } else {
+            this.dispatch.setInvalidSelectionsControlId(undefined);
+          }
+        })
+    );
```

Comment:
This is very ugly... Not sure if there could be a better way to grab the **first** control reporting invalid selections? 🙈

Reply:
Slightly improved this with ec9337725dd273736cc5bbf34344d3dd74e65ce5 based on an idea I had last night 😆  Might still be a better way though

Reply:
Oops, I created earlier and forgot to publish a draft comment before your recent change. Not sure if it is still relevant?

> Do we need an undefined BehaviorSubject for non-clearable controls? Or could we use filter and map on the getChildIds array to only return BehaviorSubjects for clearable controls?